vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html

```diff
@@ -143,33 +152,104 @@
             function onMouseLeave(e) {
                 $(this).removeClass("hova");
             }
+
+            function getSelectedSpace(blockId) {
+                $.ajax({
+                    url: "[(@{'/staff/module/'+${module.id}+'/slide/'+${slide.id}+'/space/'})]" + blockId,
+                    type: 'GET',
+                    success: function(result) {
+                    	if (result != null && result.space != null && result.space.name != null){
+                            currentSpaceName = result.space.name;
+                            $("#selectedSpace").text(currentSpaceName);
+                            $("#editSpaceBlock").text(result.title);
+                            currentSpaceId = result.space.id;
+                            if ($("#editSpaceName").find("option[value='" + currentSpaceId + "']").length == 0) {
+                                var newOption = new Option(currentSpaceName, currentSpaceId, true, true);
+                                $("#editSpaceName").append(newOption).trigger('change');
+                            } else {
+                                $("#editSpaceName").val(currentSpaceId).trigger('change');
+                            }
```

Comment:
i'm so confused, what is this if block for?

Reply:
This if block checks whether a specific option (with a value equal to currentSpaceId) already exists in the #editSpaceName dropdown menu. If the option is missing, a new one is created with currentSpaceName as the display text and currentSpaceId as the value.